utilities/test_suite/HOST/Tensor_voxel_host.cpp

```diff
@@ -0,0 +1,821 @@
+    {
+        descriptorPtr3D->layout = RpptLayout::NCDHW;
+        descriptorPtr3D->dims[0] = noOfImages;
+        descriptorPtr3D->dims[1] = numChannels;
+        descriptorPtr3D->dims[2] = maxZ;
+        descriptorPtr3D->dims[3] = maxY;
+        descriptorPtr3D->dims[4] = maxX;
+    }
+
+    descriptorPtr3D->strides[0] = descriptorPtr3D->dims[1] * descriptorPtr3D->dims[2] * descriptorPtr3D->dims[3] * descriptorPtr3D->dims[4];
+    descriptorPtr3D->strides[1] = descriptorPtr3D->dims[2] * descriptorPtr3D->dims[3] * descriptorPtr3D->dims[4];
+    descriptorPtr3D->strides[2] = descriptorPtr3D->dims[3] * descriptorPtr3D->dims[4];
+    descriptorPtr3D->strides[3] = descriptorPtr3D->dims[4];
+    descriptorPtr3D->strides[4] = 1;
+}
+
+// reads nifti-1 header file
+static int read_nifti_header_file(char* const header_file, nifti_1_header *niftiHeader)
+{
+    nifti_1_header hdr;
+
+    // open and read header
+    FILE *fp = fopen(header_file,"r");
+    if (fp == NULL)
+    {
+        fprintf(stderr, "\nError opening header file %s\n", header_file);
+        exit(1);
+    }
+    int ret = fread(&hdr, MIN_HEADER_SIZE, 1, fp);
+    if (ret != 1)
+    {
+        fprintf(stderr, "\nError reading header file %s\n", header_file);
+        exit(1);
+    }
+    fclose(fp);
+
+    // print header information
+    fprintf(stderr, "\n%s header information:", header_file);
+    fprintf(stderr, "\nNIFTI1 XYZT dimensions: %d %d %d %d", hdr.dim[1], hdr.dim[2], hdr.dim[3], hdr.dim[4]);
+    fprintf(stderr, "\nNIFTI1 Datatype code and bits/pixel: %d %d", hdr.datatype, hdr.bitpix);
+    fprintf(stderr, "\nNIFTI1 Scaling slope and intercept: %.6f %.6f", hdr.scl_slope, hdr.scl_inter);
+    fprintf(stderr, "\nNIFTI1 Byte offset to data in datafile: %ld", (long)(hdr.vox_offset));
+    fprintf(stderr, "\n");
+
+    *niftiHeader = hdr;
+
+    return(0);
+}
+
+// reads nifti-1 data file
+inline void read_nifti_data_file(char* const data_file, nifti_1_header *niftiHeader, NIFTI_DATATYPE *data)
+{
+    nifti_1_header hdr = *niftiHeader;
+    int ret;
+
+    // open the datafile, jump to data offset
+    FILE *fp = fopen(data_file, "r");
+    if (fp == NULL)
+    {
+        fprintf(stderr, "\nError opening data file %s\n", data_file);
+        exit(1);
+    }
+    ret = fseek(fp, (long)(hdr.vox_offset), SEEK_SET);
+    if (ret != 0)
+    {
+        fprintf(stderr, "\nError doing fseek() to %ld in data file %s\n", (long)(hdr.vox_offset), data_file);
+        exit(1);
+    }
+
+    ret = fread(data, sizeof(NIFTI_DATATYPE), hdr.dim[1] * hdr.dim[2] * hdr.dim[3], fp);
+    if (ret != hdr.dim[1] * hdr.dim[2] * hdr.dim[3])
+    {
+        fprintf(stderr, "\nError reading volume 1 from %s (%d)\n", data_file, ret);
+        exit(1);
+    }
+    fclose(fp);
+}
+
+inline void write_nifti_file(nifti_1_header *niftiHeader, NIFTI_DATATYPE *niftiData, int batchCount)
+{
+    nifti_1_header hdr = *niftiHeader;
+    // nifti1_extender pad = {0,0,0,0};
```

Comment:
lots of commented code still in this file. Please consolidate and remove any unwanted code